cli/cmd/create_project.go

```diff
@@ -127,9 +127,9 @@ keptn create project PROJECTNAME --shipyard=FILEPATH --git-user=GIT_USER --git-r
 				return errors.New("Access token or private key must be set")
 			}
 
-			project.GitUser = *createProjectParams.GitUser
-			project.GitToken = *createProjectParams.GitToken
-			project.GitRemoteURL = *createProjectParams.RemoteURL
+			if *createProjectParams.GitToken != "" && *createProjectParams.GitPrivateKey != "" {
```

Comment:
to be safe I suggest to use the `isStringFlagSet(s *string)` method (available in the same package in the CLI), which also checks if the passed string pointer is `nil`.